vllm/v1/worker/block_table.py

```diff
@@ -42,18 +40,19 @@ def __init__(
 
     def append_row(
         self,
-        row_idx: int,
-        start: int,
         block_ids: List[int],
+        row_idx: int,
     ) -> None:
         if not block_ids:
             return
         num_blocks = len(block_ids)
+        start = self.num_blocks_per_row[row_idx]
```

Comment:
I'm not very familiar with this. Does this always equal to `len(req_state.block_ids) - len(req_data.new_block_ids)` if not 0?

Reply:
I think so, and added some tests for the checking `block_table` is the same as `req_state.block_ids`